src/main/java/crawlercommons/sitemaps/SiteMapParser.java

```diff
@@ -265,6 +267,78 @@ public AbstractSiteMap parseSiteMap(String contentType, byte[] content, URL url)
         throw new UnknownFormatException("Can't parse a sitemap with the MediaType of: " + contentType + " (at: " + url + ")");
     }
 
+    /**
+     * Fetch a sitemap from the specified URL, recursively fetching and
+     * traversing the content of any enclosed sitemap index, and performing the
+     * specified action for each sitemap URL until all URLs have been processed
+     * or the action throws an exception.
+     * <p>
+     * This method is a convenience method for a user who has a sitemap URL and
+     * wants a simple way to traverse it.
+     * <p>
+     * Exceptions thrown by the action are relayed to the caller.
+     *
+     * @param onlineSitemapUrl
+     *            URL of the online sitemap
+     * @param action
+     *            The action to be performed for each element
+     * @throws UnknownFormatException
+     *             if there is an error parsing the sitemap
+     * @throws IOException
+     *             if there is an error fetching the content of any
+     *             {@link java.net.URL}
+     */
+    public void walkSiteMap(URL onlineSitemapUrl, Consumer<SiteMapURL> action) throws UnknownFormatException, IOException {
+        if (action == null) {
+            return;
```

Comment:
A comment using an slf4j logger will be appreciated here

Reply:
Done.

I would usually throw a `NullPointerException` in this context, but I noticed that methods like `SiteMapParser#parseSiteMap(URL)` are null-safe, hence the early return statement, for the sake of consistency.